File: backend/app/application/dto/agent.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, EmailStr


class AgentCreate(BaseModel):
    company_name: str
    contact_name: str
    contact_email: EmailStr | None = None
    note: str | None = None


class AgentUpdate(BaseModel):
    company_name: str | None = None
    contact_name: str | None = None
    contact_email: EmailStr | None = None
    note: str | None = None


class AgentResponse(BaseModel):
    id: UUID
    company_name: str
    contact_name: str
    contact_email: EmailStr | None
    note: str | None
    created_at: datetime
    updated_at: datetime


class AgentStats(BaseModel):
    id: UUID
    company_name: str
    contact_name: str
    referral_count: int
    stage_0_5_pass_rate: float
    final_offer_rate: float
    mismatch_rate: float

File: frontend/src/domain/entities/agent.ts
export interface Agent {
  id: string;
  company_name: string;
  contact_name: string;
  contact_email: string | null;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface AgentCreate {
  company_name: string;
  contact_name: string;
  contact_email?: string;
  note?: string;
}

export interface AgentUpdate {
  company_name?: string;
  contact_name?: string;
  contact_email?: string;
  note?: string;
}

export interface AgentStats {
  id: string;
  company_name: string;
  contact_name: string;
  referral_count: number;
  stage_0_5_pass_rate: number;
  final_offer_rate: number;
  mismatch_rate: number;
}

